internal/provider/resource_user_alias.go

```diff
@@ -0,0 +1,243 @@
+	return &schema.Resource{
+		Description:   "User alias resource manages individual aliases for the given Google workspace account.",
+		CreateContext: resourceUserAliasCreate,
+		ReadContext:   resourceUserAliasRead,
+		UpdateContext: nil,
+		DeleteContext: resourceUserAliasDelete,
+		Importer: &schema.ResourceImporter{
+			State: resourceUserAliasImport,
+		},
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(5 * time.Minute),
+		},
+		Schema: map[string]*schema.Schema{
+			"primary_email": {
+				Type:        schema.TypeString,
+				Description: "Primary Email (userKey) of the user the alias should be applied to.",
+				Required:    true,
+				ForceNew:    true,
+			},
+			"alias": {
+				Type:        schema.TypeString,
+				Description: "Email alias which should be applied to the user.",
+				Required:    true,
+				ForceNew:    true,
+			},
+			"etag": {
+				Description: "ETag of the resource.",
+				Type:        schema.TypeString,
+				Computed:    true,
+			},
+		},
+	}
+}
+
+func resourceUserAliasCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	var diags diag.Diagnostics
+
+	client := meta.(*apiClient)
+
+	directoryService, diags := client.NewDirectoryService()
+	if diags.HasError() {
+		return diags
+	}
+
+	usersService, diags := GetUsersService(directoryService)
+	if diags.HasError() {
+		return diags
+	}
+
+	aliasesService, diags := GetUserAliasService(usersService)
+	if diags.HasError() {
+		return diags
+	}
+
+	primaryEmail := d.Get("primary_email").(string)
+	setAlias := d.Get("alias").(string)
+
+	alias := &admin.Alias{
+		Alias: setAlias,
+	}
+	alias, err := aliasesService.Insert(primaryEmail, alias).Do()
+	if err != nil {
+		return diag.Errorf("[ERROR] failed to add alias for user (%s): %v", primaryEmail, err)
+	}
+
+	// Using a different backoff style due to the fact the top level ETag is not a consistent value for individual aliases. In theory it will change when multiple
+	// instances of this resource are used with the same primary email. The workaround is to wait until the list aliases call returns the actual alias created by the
+	// specific instance of the module. Since there is no real update lifecycle for this type of thing, the ETag really is just there for self validation nothing changed,
+	// in practice it should not matter too much.
+	bOff := backoff.NewExponentialBackOff()
+	bOff.MaxElapsedTime = d.Timeout(schema.TimeoutCreate)
+	bOff.InitialInterval = time.Second
+
+	err = backoff.Retry(func() error {
+		resp, err := aliasesService.List(primaryEmail).Do()
+		if err != nil {
+			return backoff.Permanent(fmt.Errorf("[ERROR] could not retrieve aliases for user (%s): %v", primaryEmail, err))
+		}
+
+		_, ok := doesAliasExist(resp, setAlias)
+		if ok {
+			return nil
+		}
+		return fmt.Errorf("[WARN] no matching alias (%s) found for user (%s).", setAlias, primaryEmail)
+
+	}, bOff)
+
+	d.SetId(fmt.Sprintf("%s/%s", primaryEmail, alias.Alias))
```

Comment:
@megan07 I don't believe we can use `Id` for this, per the docs (https://developers.google.com/admin-sdk/directory/v1/guides/manage-user-aliases) and some light testing of my own; this is the `Id` of the primary user account, therefore it is not unique for each alias.

Reply:
Sure. Sorry, I was thinking the same format, only `<alias.Id>/<alias.Alias>` if that makes sense? I'm thinking only because `alias.Id` is equivalent to `user.Id`, and we use `user.Id` as the id of the user, so consistency-wise I'm thinking `alias.Id` would make more sense to use in the alias  id (while still specifying the alias to keep it unique).
I think I'd prefer that, but I don't hold to that opinion too strongly. I'm curious your thoughts from a user perspective what would be the better UX.

Reply:
@megan07 This is a bit of a tough one, I think for `normal` use cases the `Id` field is the better option. However for import I lean towards `primaryEmail` since that's what "most" administrators will be aware of as the unique ID for a workspace user. 

Since you're using `Id` for users, I think I'll switch to that just as a consistency thing.

Reply:
Wanted to double-check if I should expect a change here. Thanks!

Reply:
Hi @tgoodsell-tempus, I wanted to touch base and see if you know if this is the same functionality, but different API: https://github.com/hashicorp/terraform-provider-googleworkspace/pull/122/files
Would it work for what you're looking for? Thanks!